supervisord.py

```diff
@@ -47,14 +48,29 @@ def check(self, instance):
             AgentCheck.UNKNOWN: 0
         }
 
-        # Report service checks and uptime for each process
+        proc_regex = instance.get('proc_regex')
         proc_names = instance.get('proc_names')
+        processes = server.supervisor.getAllProcessInfo()
+        monitored_processes = []
+
+        # default to all processes if no process filters are configured
+        if not proc_regex and not proc_names:
+            monitored_processes = processes
+
+        # Add regex matches to process list
+        if proc_regex and len(proc_regex):
+            patterns = '(?:%s)' % '|'.join(proc_regex)
+            for process in processes:
+                if re.match(patterns, process) and process not in monitored_processes:
+                    monitored_processes.append(server.supervisor.getProcessInfo(process))
+
+        # Add explicit matches to process list
         if proc_names and len(proc_names):
-            processes = [server.supervisor.getProcessInfo(p) for p in proc_names]
-        else:
-            processes = server.supervisor.getAllProcessInfo()
+            for process in proc_names:
+                if process not in monitored_processes:
```

Comment:
this seems off...   isn't `monitored_processes` a collection of (*what i'm assuming is*) `ProcessInfo` objects? Whereas `process` here... is (*what I'm assuming*) is just a string obtained from `proc_names`?

maybe i need to read up on the docs a bit more...   what do `getAllProcessInfo()` and `getProcessInfo()` return?